exec-healthz/Makefile

```diff
@@ -1,17 +1,70 @@
-all: push
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Build the exechealthz binary or image for any GOARCH
+#
+# Usage:
+# 	[TAG=1.0] [REGISTRY=gcr.io/google_containers] [ARCH=amd64] make [server|container|push]
+
+all: container
 
 # 0.0 shouldn't clobber any released builds
-TAG = 0.0
-PREFIX = gcr.io/google_containers/exechealthz
+TAG?=1.0
+REGISTRY?=gcr.io/google_containers
+ARCH?=amd64
+GOLANG_VERSION=1.6
+TEMP_DIR:=$(shell mktemp -d)
+
+# Set default base image dynamically for each arch
+ifeq ($(ARCH),amd64)
+	BASEIMAGE?=busybox
+endif
+ifeq ($(ARCH),arm)
+	BASEIMAGE?=hypriot/armhf-busybox
+endif
+ifeq ($(ARCH),arm64)
+	BASEIMAGE?=aarch64/busybox
+endif
+ifeq ($(ARCH),ppc64le)
+	BASEIMAGE?=ppc64le/busybox
+endif
 
 server: exechealthz.go
-	CGO_ENABLED=0 GOOS=linux godep go build -a -installsuffix cgo -ldflags '-w' -o exechealthz ./exechealthz.go
+	CGO_ENABLED=0 GOOS=linux GOARCH=$(ARCH) GOARM=6 go build -a -installsuffix cgo -ldflags '-w' -o exechealthz ./exechealthz.go
+
+container:
+	# Copy the whole directory to a temporary dir and set the base image
```

Comment:
Should we switch to `go install` so that we can cache past build results? And instead of cp-ing the entire directory create a `Dockerfile-arch` (which would be in .gitignore) and build the container from there?